TinYard/Framework/Impl/Context.cs

```diff
@@ -32,31 +34,36 @@ public class Context : IContext
         private object _environment;
 
         //Private variables
-        private List<IExtension> _extensionsToInstall;
+        private readonly ILogger<Context> _logger;
+
+        private readonly List<IExtension> _extensionsToInstall;
         private HashSet<IExtension> _extensionsInstalled;
 
-        private HashSet<IBundle> _bundlesInstalled;
+        private readonly HashSet<IBundle> _bundlesInstalled;
 
-        private List<IConfig> _configsToInstall;
+        private readonly List<IConfig> _configsToInstall;
         private HashSet<IConfig> _configsInstalled;
 
-        private HashSet<object> _detainedObjs;
+        private readonly HashSet<object> _detainedObjs;
 
         private bool _initialized = false;
 
-        public Context()
+        public Context( ILoggerFactory loggerFactory = null)
         {
+            _logger = loggerFactory?.CreateLogger<Context>();
```

Comment:
What happens if loggerFactory is null. Surely all _logger calls would fail? (Question, not sure myself)

Reply:
I believe that the `ILogger` extensions I've made should handle this. I'll test though as I'm not 100% sure either

Reply:
Ok, I've tested this. It works fine 👍🏼